src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,881 @@
+        When ad_gpo_access_control is set to disabled, no policy processing should occur and be
+        turned off entirely.
+    :setup:
+        1. Create the following users; 'user', 'user1' and 'deny_user1'
+        2. Create and link the GPO 'site policy' and add 'user1', group and 'Domain Admins' to
+           SeInteractiveLogonRight key, and add 'deny_user1' to SeDenyInteractiveLogonRight key
+        3. Configure sssd.conf with 'ad_gpo_access_control' = 'disabled'
+        4. Start SSSD
+    :steps:
+        1. Authenticate as 'user', 'user1' and 'deny_user1'
+        2. Check logs for ad_gpo_access_control value
+    :expectedresults:
+        1. All users authentication are successful
+        2. ad_gpo_access_control is disabled
+    :customerscenario: True
+    """
+    ad.user("user").add()
+    user1 = ad.user("user1").add()
+    deny_user1 = ad.user("deny_user1").add()
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "disabled"
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password(username="user", password="Secret123")
+    assert client.auth.parametrize(method).password(username="user1", password="Secret123")
+    assert client.auth.parametrize(method).password(username="deny_user1", password="Secret123")
+
+    log_str = client.fs.read(client.sssd.logs.domain())
+    assert "Option ad_gpo_access_control has value disabled" in log_str
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["ssh", "su"])
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.ticket(bz=1695576)
+def test_gpo__implicit_deny_is_set_to_true(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing and implicit deny is true
+    :description:
+        When ad_gpo_implicit_deny is set to true, the new default behavior is to deny access for
+        when there is no policy applied.
+    :setup:
+        1. Create the following users; 'user'
+        2. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing' and 'ad_gpo_implicit_deny' = True
+        3. Start SSSD
+    :steps:
+        1. Authenticate as 'user'
+    :expectedresults:
+        1. 'user' authentication is unsuccessful
+    :customerscenario: True
+    """
+    ad.user("user").add()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.domain["ad_gpo_implicit_deny"] = "True"
+    client.sssd.start()
+
+    assert not client.auth.parametrize(method).password(username="user", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["ssh", "su"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__domain_and_sites_inheritance_when_site_is_enforcing(client: Client, ad: AD, method: str):
```

Comment:
Hi,

maybe you can add the same test but enforce the domain policy to show that the domain policy is prefrerred in this case.

bye,
Sumit

Reply:
The domain policy is always preferred over sites, the next test case **test_gpo__domain_and_sites_inheritance** tests that.

Reply:
Ah, thanks, you are right, I mixed up the order between site and domain.